flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/internal/executor/TableBufferedStatementExecutor.java

```diff
@@ -52,11 +52,13 @@ public void addToBatch(RowData record) throws SQLException {
 
     @Override
     public void executeBatch() throws SQLException {
-        for (RowData value : buffer) {
-            statementExecutor.addToBatch(value);
+        if (!buffer.isEmpty()) {
+            for (RowData value : buffer) {
+                statementExecutor.addToBatch(value);
+            }
+            statementExecutor.executeBatch();
+            buffer.clear();
         }
-        statementExecutor.executeBatch();
-        buffer.clear();
     }
```

Comment:
Would it make sense to return early if buffer is empty in both methods? (It would show the change clearly)

Reply:
+1

Reply:
From one side yes from another side current change is done in a way of already existing style like e.g. similar existing code for 
`org.apache.flink.connector.jdbc.internal.executor.InsertOrUpdateJdbcExecutor#executeBatch`
may be we need to refactor all of them at some point